core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -701,20 +708,23 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
         fastPassServiceConstructor(childContext, dataSource).removeFastPassGrantsForWorkspace(workspaceContext)
       )
 
+      // notify leonardo so it can cleanup any dangling sam resources and other non-cloud state
+      _ <- traceFutureWithParent("notifyLeonardo", parentContext)(_ =>
+        leonardoService.cleanupResources(workspaceContext.googleProjectId, workspaceContext.workspaceIdAsUUID, ctx)
+      )
+
       // Delete Google Project
       _ <- traceFutureWithParent("maybeDeleteGoogleProject", parentContext)(_ =>
-        if (!isAzureMcWorkspace(maybeMcWorkspace)) {
-          maybeDeleteGoogleProject(workspaceContext.googleProjectId,
-                                   workspaceContext.workspaceVersion,
-                                   ctx.userInfo
-          ) recoverWith { case t: Throwable =>
-            logger.error(
-              s"Unexpected failure deleting workspace (while deleting google project) for workspace `${workspaceContext.toWorkspaceName}`",
-              t
-            )
-            Future.failed(t)
-          }
-        } else Future.successful()
+        maybeDeleteGoogleProject(workspaceContext.googleProjectId,
```

Comment:
This is the only difference from the original PR. We're notifying leo to cleanup resources first, then deleting the gcp project and related sam gcp project resource after, rather than splitting the operations.

Reply:
I did not realize that it could take up to 24h to delete a project, in this case making the delete project call before is not that important

Reply:
And I see `deletePet` in that method... https://github.com/broadinstitute/rawls/blob/456fbf19de5407006cdeba6f966dfd8d1298ab18/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L837